pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -394,6 +394,10 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.I
 		PortRange:  fmt.Sprintf("%d-%d", minPort, maxPort),
 		Target:     gce.targetPoolURL(name, region),
 	}
+	if loadBalancerIP != nil {
+		req.IPAddress = loadBalancerIP.String()
```

Comment:
It uses the `externalIP` arg passed to `EnsureTCPLoadBalancer` (if non-nil).

So there's two cases (as I understand the existing code):
- type=LoadBalancer and no ExternalIPs:  the plugin allocates a new public IP, uses it, and destroys it when you remove the service/loadbalancer.
- type=LoadBalancer and ExternalIPs: the plugin is passed the first externalIP and the openstack plugin just uses it as the loadbalancer IP.  If the user provided an invalid IP for some reason, it will return an error.  How/where the user got that IP is up to them.